internal/promotion/runner/builtin/git_pr_opener.go

```diff
@@ -75,17 +75,23 @@ func (g *gitPROpener) run(
 ) (promotion.StepResult, error) {
 	// Short-circuit if shared state has output from a previous execution of this
 	// step that contains a PR number.
+	//
+	// TODO(krancour): This short-circuiting logic may have outlived its
+	// usefulness now that we keep track of the current step.
```

Comment:
@fykaa after further consideration and a conversation with @hiddeco, I would indeed like to see this short-circuiting logic removed.